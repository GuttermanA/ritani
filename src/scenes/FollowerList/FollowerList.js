import React, { Component, Fragment } from "react";
import { MouseOverAvatar, LoadFollowersButton } from "./components";
import { Container } from "components";
import uuid from "uuid/v4";

class FollowerList extends Component {
  render() {
    const {
      followerData,
      children,
      lastPage,
      page,
      loadMoreFollowers,
      fetchUserWithFollowers,
    } = this.props;
    const followers =
      followerData &&
      followerData.map(follower => {
        const { avatar_url, login, url, followers_url } = follower;
        return (
          <MouseOverAvatar
            src={avatar_url}
            alt={`${login} follower avatar`}
            id={login}
            url={url}
            followersURL={followers_url}
            key={uuid()}
            fetchUserWithFollowers={fetchUserWithFollowers}
          />
        );
      });

    return (
      <Fragment>
        <Container className="container">{followers}</Container>
        <LoadFollowersButton
          lastPage={lastPage}
          page={page}
          loadMoreFollowers={loadMoreFollowers}
        />
        {children}
      </Fragment>
    );
  }
}

export default FollowerList;
