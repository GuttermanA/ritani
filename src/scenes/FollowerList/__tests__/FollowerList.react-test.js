import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import FollowerList from "../FollowerList";

const mockFollowers = [
  {
    login: "Vkopensource",
    id: 48538853,
    node_id: "MDQ6VXNlcjQ4NTM4ODUz",
    avatar_url: "https://avatars2.githubusercontent.com/u/48538853?v=4",
    gravatar_id: "",
    url: "https://api.github.com/users/Vkopensource",
    html_url: "https://github.com/Vkopensource",
    followers_url: "https://api.github.com/users/Vkopensource/followers",
    following_url:
      "https://api.github.com/users/Vkopensource/following{/other_user}",
    gists_url: "https://api.github.com/users/Vkopensource/gists{/gist_id}",
    starred_url:
      "https://api.github.com/users/Vkopensource/starred{/owner}{/repo}",
    subscriptions_url:
      "https://api.github.com/users/Vkopensource/subscriptions",
    organizations_url: "https://api.github.com/users/Vkopensource/orgs",
    repos_url: "https://api.github.com/users/Vkopensource/repos",
    events_url: "https://api.github.com/users/Vkopensource/events{/privacy}",
    received_events_url:
      "https://api.github.com/users/Vkopensource/received_events",
    type: "User",
    site_admin: false
  },
  {
    login: "stvtortora",
    id: 32438308,
    node_id: "MDQ6VXNlcjMyNDM4MzA4",
    avatar_url: "https://avatars3.githubusercontent.com/u/32438308?v=4",
    gravatar_id: "",
    url: "https://api.github.com/users/stvtortora",
    html_url: "https://github.com/stvtortora",
    followers_url: "https://api.github.com/users/stvtortora/followers",
    following_url:
      "https://api.github.com/users/stvtortora/following{/other_user}",
    gists_url: "https://api.github.com/users/stvtortora/gists{/gist_id}",
    starred_url:
      "https://api.github.com/users/stvtortora/starred{/owner}{/repo}",
    subscriptions_url: "https://api.github.com/users/stvtortora/subscriptions",
    organizations_url: "https://api.github.com/users/stvtortora/orgs",
    repos_url: "https://api.github.com/users/stvtortora/repos",
    events_url: "https://api.github.com/users/stvtortora/events{/privacy}",
    received_events_url:
      "https://api.github.com/users/stvtortora/received_events",
    type: "User",
    site_admin: false
  }
];

describe("<FollowerList/>", () => {
  test("renders correctly", () => {
    const component = render(<FollowerList />);
    expect(component).toMatchSnapshot();
  });

  test("renders followers passed as a prop", () => {
    const { queryAllByTestId } = render(
      <FollowerList followerData={mockFollowers} />
    );

    expect(queryAllByTestId("mouse-over-avatar-member").length).toBe(2);
  });
});
