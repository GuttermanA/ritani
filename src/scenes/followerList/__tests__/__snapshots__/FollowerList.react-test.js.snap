// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FollowerList/> renders correctly 1`] = `
Array [
  <div
    className="container"
  />,
  <button
    className="button centered"
    type="submit"
  >
    Load
  </button>,
  <div />,
]
`;

exports[`<FollowerList/> renders followers passed as a prop 1`] = `
Array [
  <div
    className="container"
  >
    <div
      className="avatar-container"
      data-testid="avatar-container"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <img
        alt="Vkopensource follower avatar"
        className="avatar"
        data-testid="follower-avatar"
        src="https://avatars2.githubusercontent.com/u/48538853?v=4"
      />
    </div>
    <div
      className="avatar-container"
      data-testid="avatar-container"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <img
        alt="stvtortora follower avatar"
        className="avatar"
        data-testid="follower-avatar"
        src="https://avatars3.githubusercontent.com/u/32438308?v=4"
      />
    </div>
  </div>,
  <button
    className="button centered"
    type="submit"
  >
    Load
  </button>,
  <div />,
]
`;
